crates/services/bls_aggregation/src/bls_agg.rs

```diff
@@ -904,6 +958,122 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
         }
         true
     }
+
+    /// Checks if the signature is a duplicate.
+    ///
+    /// # Arguments
+    ///
+    /// * `aggregated_operators` - The aggregated operators.
+    /// * `signed_digest` - The signed task response digest.
+    ///
+    /// # Returns
+    ///
+    /// Returns `true` if the signature is a duplicate, otherwise `false`.
+    fn is_duplicate_signature(
+        aggregated_operators: &HashMap<FixedBytes<32>, AggregatedOperators>,
+        signed_digest: &SignedTaskResponseDigest,
+    ) -> bool {
+        aggregated_operators
+            .get(&signed_digest.task_response_digest)
+            .map(|ops| {
+                ops.signers_operator_ids_set
+                    .contains_key(&signed_digest.operator_id)
+            })
+            .unwrap_or(false)
+    }
+
+    /// Updates the aggregated operators with the new operator info.
+    ///
+    /// # Arguments
+    ///
+    /// * `aggregated_operators` - The aggregated operators.
+    /// * `operator_state` - The operator state.
+    /// * `task_response_digest` - The task response digest.
+    /// * `bls_signature` - The BLS signature.
+    /// * `operator_id` - The operator id.
+    /// * `logger` - The logger to log messages.
+    ///
+    /// # Returns
+    ///
+    /// The updated aggregated operators.
+    fn update_aggregated_operators(
+        aggregated_operators: &mut HashMap<FixedBytes<32>, AggregatedOperators>,
+        operator_state: &OperatorAvsState,
+        task_response_digest: FixedBytes<32>,
+        bls_signature: Signature,
+        operator_id: FixedBytes<32>,
+        logger: SharedLogger,
+    ) -> AggregatedOperators {
+        logger.debug(
+            "Update aggregated operators",
+            "eigen-services-blsaggregation.bls_agg.update_aggregated_operators",
+        );
+
+        let bls_signature_g1_point = bls_signature.g1_point().g1();
+
+        if let Some(existing) = aggregated_operators.get_mut(&task_response_digest) {
+            // If the operator is already in the aggregated operators, aggregate the new operator
+            let updated = Self::aggregate_new_operator(
+                existing,
+                operator_state.clone(),
+                operator_id,
+                bls_signature_g1_point,
+                logger,
+            );
+            updated.clone()
```

Comment:
```suggestion
            Self::aggregate_new_operator(
                existing,
                operator_state.clone(),
                operator_id,
                bls_signature_g1_point,
                logger,
            )
```

Reply:
Nvm. Let's leave it like this then. We should later check if there's a way to avoid cloning altogether. Mostly because it's weird and not so much for performance reasons.